### PR TITLE
Bluetooth: Mesh: Point to point and Mesh DFU

### PR DESCRIPTION
**Pull request by Balaklaka**
Adding point to point DFU together with Mesh DFU
so the point to point DFU can be used as a fallback solution.

### CODE REVIEW

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -279,6 +279,7 @@ Bluetooth Mesh samples
 * :ref:`ble_mesh_dfu_target` sample:
 
   * Added a note on how to compile the sample with new Composition Data.
+  * Added point-to-point DFU support by adding :file:`overlay-ptp_dfu.conf` overlay.
```
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:282]:
```suggestion
  * Added point-to-point DFU support with overlay file :file:`overlay-ptp_dfu.conf`.
```

File: samples/bluetooth/mesh/dfu/target/README.rst [hunk 2]
```diff
@@ -29,6 +29,13 @@ For uploading an image to the Distributor, this sample also requires a smartphon
 * `nRF Device Manager mobile app for Android`_
 * `nRF Device Manager mobile app for iOS`_
 
+Point-to-point DFU requirements
+*******************************
+
+The configuration overlay :file:`overlay-ptp_dfu.conf` enables the :ref:`dfu_over_ble` feature.
+
+This is added as a backup option for the DFU process in case the mesh network is not available.
```
Comment (mia-ko) [inline samples/bluetooth/mesh/dfu/target/README.rst:37]:
```suggestion
This feature can be used as a backup option for the DFU process if the mesh network is not available.
```

**Review by mia-ko — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -279,6 +279,7 @@ Bluetooth Mesh samples
 * :ref:`ble_mesh_dfu_target` sample:
 
   * Added a note on how to compile the sample with new Composition Data.
+  * Added point-to-point DFU support with overlay file :file:`overlay-ptp_dfu.conf`.
```
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:282]:
```suggestion
  * Added:
  
    * A note on how to compile the sample with new Composition Data.
    * Point-to-point DFU support with overlay file :file:`overlay-ptp_dfu.conf`.
```

**Review by mia-ko — CHANGES_REQUESTED** [review]
Just one more small change, that I failed to notice in the last round :)

**Review by mia-ko — APPROVED** [review]

File: samples/bluetooth/mesh/dfu/target/src/main.c [hunk 4]
```diff
@@ -14,6 +14,9 @@
 #include <zephyr/bluetooth/bluetooth.h>
 #include <zephyr/bluetooth/mesh.h>
 #include "dfu_target.h"
+#if IS_ENABLED(CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU)
```
Comment (alxelax) [inline samples/bluetooth/mesh/dfu/target/src/main.c:17]:
```suggestion
#if defined(CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU)
```
https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/util_macro.h#L117-L118

File: samples/bluetooth/mesh/dfu/target/src/main.c [hunk 5]
```diff
@@ -117,6 +120,14 @@ static void bt_ready(int err)
 
 	printk("Mesh initialized\n");
 
+#if IS_ENABLED(CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU)
```
Comment (alxelax) [inline samples/bluetooth/mesh/dfu/target/src/main.c:123]:
```suggestion
#if defined(CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU)
```

File: samples/bluetooth/mesh/dfu/distributor/README.rst [hunk 6]
```diff
@@ -219,6 +219,8 @@ When the Distributor updates itself, the DFU transfer will end immediately after
 
 When this sample is used as a Target, it behaves as described in :ref:`ble_mesh_dfu_target_upgrade`.
 
+This sample also provides support for :ref:`dfu_over_ble`, so it is possible to self-update using the Simple Management Protocol (SMP).
```
Comment (alxelax) [inline samples/bluetooth/mesh/dfu/distributor/README.rst:222]: I didn't get how distributor sample is involved into p2p dfu. Could you clarify?

File: samples/bluetooth/mesh/dfu/target/README.rst [hunk 7]
```diff
@@ -29,6 +29,13 @@ For uploading an image to the Distributor, this sample also requires a smartphon
 * `nRF Device Manager mobile app for Android`_
 * `nRF Device Manager mobile app for iOS`_
 
+Point-to-point DFU requirements
+*******************************
+
+The configuration overlay :file:`overlay-ptp_dfu.conf` enables the :ref:`dfu_over_ble` feature.
+
+This feature can be used as a backup option for the DFU process if the mesh network is not available.
```
Comment (alxelax) [inline samples/bluetooth/mesh/dfu/target/README.rst:37]: Could we clarify usage of them a bit? I mean impossibility to use in parallel. Or for example that, one procedure will not pick up dfu process of another if it was interrupted\suspended.

**Review by alxelax — COMMENTED** [review]
LGTM, but requires a bit more work.

File: samples/bluetooth/mesh/dfu/distributor/README.rst (hunk 6 above)
Comment (Balaklaka) [reply samples/bluetooth/mesh/dfu/distributor/README.rst:222]: The sample allready have suppport for p2p dfu, but there is no comment about that in the documentation. This is only to give the user a hint that they also can use p2p to do self update of the distributor. One of the points in the task was to test if p2p for the distributor sample was working, and it was. So then I deceiced to mention that in the sample. Did not want to give to much documentation, as I want them to read the things in dfu_over_ble.

File: samples/bluetooth/mesh/dfu/target/README.rst (hunk 7 above)
Comment (Balaklaka) [reply samples/bluetooth/mesh/dfu/target/README.rst]: @mia-ko : I changed the text here, could you check what I wrote?

File: samples/bluetooth/mesh/dfu/target/README.rst [hunk 8]
```diff
@@ -29,6 +29,14 @@ For uploading an image to the Distributor, this sample also requires a smartphon
 * `nRF Device Manager mobile app for Android`_
 * `nRF Device Manager mobile app for iOS`_
 
+Point-to-point DFU requirements
+*******************************
+
+The configuration overlay :file:`overlay-ptp_dfu.conf` enables the :ref:`dfu_over_ble` feature.
+
+This feature can be used together with Bluetooth Mesh DFU.
+If the Bluetooth Mesh DFU procedure is suspended, failing or if the Bluetooth Mesh network is not available, the point-to-point DFU feature can be used as a backup option for the DFU process.
```
Comment (mia-ko) [inline samples/bluetooth/mesh/dfu/target/README.rst:38]:
```suggestion
If the Bluetooth Mesh DFU procedure is suspended, failing, or if the Bluetooth Mesh network is not available, the point-to-point DFU feature can be used as a backup option for the DFU process.
```

**Review by mia-ko — CHANGES_REQUESTED** [review]
A tiny, tiny change.
Otherwise looks good :)

**Review by mia-ko — APPROVED** [review]

File: samples/bluetooth/mesh/dfu/distributor/README.rst (hunk 6 above)
Comment (ludvigsj) [reply samples/bluetooth/mesh/dfu/distributor/README.rst:222]: If you do it this way (same for Target), you should also create the reverse references in `dfu_over_ble.rst`. Currently it specifically mentions the Light sample and Thingy53, which makes it confusing that the DFU samples also supports it but is not listed there.

**Review by ludvigsj — CHANGES_REQUESTED** [review]

File: samples/bluetooth/mesh/dfu/distributor/README.rst (hunk 6 above)
Comment (mia-ko) [reply samples/bluetooth/mesh/dfu/distributor/README.rst:222]: Ah, that's true, it should be mentioned in https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/protocols/bt/bt_mesh/dfu_over_ble.html#point-to_point_dfu_over_bluetooth_low_energy_in_bluetooth_mesh_samples

Comment (Balaklaka) [reply samples/bluetooth/mesh/dfu/distributor/README.rst:222]: Added

**Review by alxelax — APPROVED** [review]

**Review by ludvigsj — APPROVED** [review]

File: doc/nrf/protocols/bt/bt_mesh/dfu_over_ble.rst [hunk 9]
```diff
@@ -15,8 +15,8 @@ See `Discovering Bluetooth Mesh devices in nRF Connect Device Manager`_ for more
 Point-to point DFU over Bluetooth Low Energy in Bluetooth Mesh samples
 **********************************************************************
 
-The :ref:`bluetooth_mesh_light` sample enables support for point-to-point DFU over Bluetooth Low Energy for nRF52 Series development kits.
-See the sample documentation for more details.
+The :ref:`bluetooth_mesh_light`, :ref:`ble_mesh_dfu_target` and :ref:`ble_mesh_dfu_distributor` samples enables support for point-to-point DFU over Bluetooth Low Energy for nRF52 Series development kits.
+See the samples documentation for more details.
```
Comment (mia-ko) [inline doc/nrf/protocols/bt/bt_mesh/dfu_over_ble.rst:19]:
```suggestion
The :ref:`bluetooth_mesh_light`, :ref:`ble_mesh_dfu_target` and :ref:`ble_mesh_dfu_distributor` samples enable support for point-to-point DFU over Bluetooth Low Energy for nRF52 Series development kits.
See the sample documentation for each of the above mentioned samples for more details.
```

**Review by mia-ko — CHANGES_REQUESTED** [review]

**Review by mia-ko — APPROVED** [review]

**Pull request merged**